rabbit-escape-render/src/rabbitescape/render/gameloop/WaterDynamics.java

```diff
@@ -0,0 +1,157 @@
+package rabbitescape.render.gameloop;
+
+import java.util.List;
+import java.util.Vector;
+
+import static rabbitescape.engine.CellularDirection.*;
+
+import rabbitescape.engine.WaterRegion;
+import rabbitescape.engine.World;
+import rabbitescape.engine.util.CellDebugPrint;
+import rabbitescape.engine.util.Dimension;
+import rabbitescape.engine.util.LookupTable2D;
+import rabbitescape.render.Polygon;
+import rabbitescape.render.WaterRegionRenderer;
+
+/**
+ * Manages animation of polygons for water.
+ */
+public class WaterDynamics
+{
+    public final LookupTable2D<WaterRegionRenderer> lookupRenderer ;
+    public final Vector<Polygon> polygons;
+    private int lastFramenumber = 10;
+    public final Dimension worldSize;
+
+    public WaterDynamics( World world )
+    {
+        worldSize = world.size;
+        lookupRenderer = new LookupTable2D<WaterRegionRenderer>( worldSize );
+        polygons = new Vector<Polygon>();
+    }
+
+    private WaterDynamics()
+    {
+        lookupRenderer = null;
+        polygons = null;
+        worldSize = null;
+    }
+
+    public static WaterDynamics getDummyWaterDynamics()
+    {
+        return new WaterDynamics();
+    }
+
+    /**
+     * Animation step update. Several may have passed before this is called.
+     */
+    public void update( int frameNumber )
+    {
+        if ( null == polygons )
+        {
+            return;
+        }
+        int interval = frameNumber - lastFramenumber;
+        interval = ( interval < 0 ) ? ( interval + 10 ) : ( interval );
+        lastFramenumber = frameNumber;
+        for ( int i = 0 ; i < interval ; i++ )
+        {
+            for( WaterRegionRenderer wrr: lookupRenderer )
+            {
+                wrr.setWaterHeight();
+            }
+            for( WaterRegionRenderer wrr: lookupRenderer )
+            {
+                wrr.removeHeightGaps();
+            }
+        }
+        calculatePolygons();
+    }
+
+    private void calculatePolygons()
+    {
+        polygons.clear();
+        for ( WaterRegionRenderer wrr: lookupRenderer )
+        {
+            wrr.drawnLB = wrr.drawnLT = wrr.drawnR = false;
+        }
+        for ( int y = 0; y < worldSize.height ; y++ )
+        {
+            Polygon p = new Polygon();
+            WaterRegionRenderer start = null;
+            for ( int x = 0; x < worldSize.width; x++ )
+            {
+                WaterRegionRenderer wrr = lookupRenderer.getItemAt( x, y );
+                if ( null == wrr || wrr.adjacentWaterIsFalling( HERE ) )
+                {
+                    continue;
+                }
```

Comment:
i noticed those wrr.drawn* booleans are vestigial, and removed them.

i would not be surprised if this method could be simplified, but I cant see it at the moment.